lib/models/project.js

```diff
@@ -255,12 +255,28 @@ class Project {
       let appConfig = this.require(configPath)(env);
       let addonsConfig = this.getAddonsConfig(env, appConfig);
 
-      return _.merge(addonsConfig, appConfig);
+      const mergedConfig = _.merge(addonsConfig, appConfig);
+      mergedConfig.EmberENV.DEPRECATION_COMPLIANCE = this.getDeprecationComplianceConfig();
+      return mergedConfig;
     } else {
       return this.getAddonsConfig(env, {});
     }
   }
 
+  getDeprecationComplianceConfig() {
+    const deprecationCompliance = {};
+    deprecationCompliance.self = this.pkg.ember.deprecationCompliance;
+    deprecationCompliance.addons = {};
+
+    this.addons.forEach((x) => {
+      if (x.pkg.ember && x.pkg.ember.deprecationCompliance) {
+        deprecationCompliance.addons[x.name] = x.pkg.ember.deprecationCompliance;
+      }
+    });
+
+    return deprecationCompliance;
+  }
```

Comment:
The addon hierarchy is a tree structure. As it stands, this will only grab from the top level addons. In order to traverse all addons you'll probably want to recurse.

Something like:

```js
  getDeprecationComplianceConfig() {
    const deprecationCompliance = {
      self: this.pkg.ember.deprecationCompliance,
      addons: {},
    };

    this._gatherDeprecationComplianceFromAddon(this.project, deprecationCompliance);

    return deprecationCompliance;
  }

  _gatherDeprecationComplianceFromAddon(addonOrProject, deprecationCompliance) {
    addonOrProject.addons.forEach((x) => {
      if (x.pkg.ember && x.pkg.ember.deprecationCompliance) {
        deprecationCompliance.addons[x.name] = x.pkg.ember.deprecationCompliance;
      }
    });
  }
```

(Assuming that we want to merge things into a flat list)